lib/SILOptimizer/Mandatory/MoveOnlyAddressCheckerUtils.cpp

```diff
@@ -731,6 +644,29 @@ struct UseState {
 
   SILFunction *getFunction() const { return address->getFunction(); }
 
+  /// The number of fields in the exploded type.
+  unsigned getNumSubelements() {
+    if (!cachedNumSubelements) {
+      cachedNumSubelements = TypeSubElementCount(address);
+    }
+    return *cachedNumSubelements;
+  }
+
+  void recordLivenessUse(SILInstruction *inst, SmallBitVector const &bits) {
+    auto iter = livenessUses.find(inst);
+    if (iter == livenessUses.end()) {
+      iter = livenessUses.insert({inst, SmallBitVector(getNumSubelements())})
+                 .first;
+    }
+    iter->second |= bits;
+  }
+
+  void recordLivenessUse(SILInstruction *inst, TypeTreeLeafTypeRange range) {
+    SmallBitVector bits(getNumSubelements());
+    range.setBits(bits);
```

Comment:
Not really important, but rather than allocating/setting a new bitvector here, you can just repeat the livenessUses lookup above and iterate over the range setting each bit in `iter->second`

Reply:
Fixed.